### ISSUE TITLE
bug: Platform MD ion-content wrong --offset-bottom

### ISSUE THREAD

**Issue by StefanRein** [ionic-team/ionic-framework#21130]
# Bug Report

**Ionic version:**
[x] **5.x**

**Where?**
In the Ionic 5 tabs example. 
On the **iOS** theme this **works**. 
On the **MD** theme it **does not work**.

**Current behavior:**
`<app-explore-container>` has a wrong y-position in the tab, which was initially loaded.
The `<ion-content>` has on reload of the current tab a wrong offset of `--offset-bottom:0px;`

**Expected behavior:**
The `<app-explore-container>` component should have the same y-position on every tab on the MD (Material Design) theme.
<ion-content> should have an offset of `--offset-bottom:57px;` on reload of the current tab, as the other tabs have, if you switch to some other tab.

**Steps to reproduce:**
Start a new Ionic 5 Tab Project and open Chrome with the Pixel 2 responsive emulator and switch between the tabs. The content position will change.

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)             : 4.10.2 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.0.7
   @angular-devkit/build-angular : 0.803.26
   @angular-devkit/schematics    : 8.3.26
   @angular/cli                  : 8.3.26
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : ios 5.1.1
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.0, (and 4 other plugins)

System:

   Android SDK Tools : 25.2.3 (/Users/stefanrein/Library/Android/sdk)
   ios-sim           : 8.0.2
   NodeJS            : v10.15.3 (/usr/local/bin/node)
   npm               : 6.14.3
   OS                : macOS
   Xcode             : Xcode 11.4 Build version 11E146


```

Example, when the Tab1 URL was initially loaded: http://localhost:4200/tabs/tab1

![image](https://user-images.githubusercontent.com/4970737/80283007-f6ec5300-8714-11ea-844d-22eb9180e422.png)

https://github.com/ionic-team/starters/tree/master/angular/official/tabs

**Comment by Cloud7050** [comment]
Same issue here. https://postimg.cc/N975hbYg

**Comment by StefanRein** [comment]
# Reproduction

Please follow these steps to reproduce (from https://ionicframework.com/docs/intro/cli):

```
npm uninstall -g ionic
npm install -g @ionic/cli

ionic start myApp tabs

// Choose: ❯ Angular | https://angular.io 
// ? Integrate your new app with Capacitor to target native iOS and Android? (y/N) 
// Choose: N

cd myApp/
ionic serve

// > ng run app:serve --host=localhost --port=8100
```

Now on macOS Catalina it opens the latest **Chrome Browser** (Version 81.0.4044.129)

The MD Mode is active.

1. Open the Developer Tools (Alt + CMD + i)

        Switching between tabs: It works as expected.

2. Chrome: Toggle Device Toolbar (the Mobile / Tablet icon)

3. Select iPhone X and refresh the page (CMD + r)

        It has changed the mode to iOS.

4. Switch between the tabs: It works as expected.

5. Select Pixel 2 XL and refresh the page (CMD + r)

        It has changed the mode to MD.

Switch between the tabs: It does not work as expected. 
The content jumps like I showed in the screenshots above if you draw a horizontal line between the text.

---

Do you need further information? 
Should I clone your repository into another GitHub account, so you can git clone it? 
Would this be easier for you?

I am happy to provide you more information.

**Comment by moganarangan** [comment]
I did this - 
`ionic start my-app tabs --type=angular`

`Ionic serve`

By default the tabs loads first having this offset issue.

The default opening tab have `--offset-bottom:0px;`

Other tabs have `--offset-bottom:56px;`

**Comment by Boosten** [comment]
@liamdebeasi my guess is that these issues al relate to each other: #21351 #21940 #20614

**Comment by digital-pollution** [comment]
have come across this issue also, is there a temporary workaround?

**Comment by 1x2x3x4x** [comment]
This isssue is still happening.

**Comment by SvenBudak** [comment]
after 3 years... any news? i got the same issue. happens when i have 2x ion-toolbar inside my ion-footer.

![image](https://user-images.githubusercontent.com/4930169/223296983-e54871f3-ea89-4a60-ada1-b9c1bc0313f6.png)

**Comment by SvenBudak** [comment]
I found a dirty solution:

```
setTimeout(() => {
  // Hack to force ion-content to update resize.
  // https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/content/content.tsx#L121
  window.dispatchEvent(new Event('appload'));
}, 250);
```

Suuuuper ultra mega duper dirty. but currently the only solution i found for this. Was not easy to find, because for some reason the Ionic Team dont make this resize stuff public.

**Comment by SvenBudak** [comment]
My previous recommendation unfortunately did not work cleanly in the long run. I have a better suggestion here. I've been using it myself since yesterday and have not had any more problems with incorrectly sized ion-content since then:

```ts
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class LifecycleService {
  viewDidEnter$ = new Subject<void>();
  viewDidLeave$ = new Subject<void>();
}
```

app.component.ts
```ts
constructor(private lifecycleService: LifecycleService) {}

ngOnInit() {
  this.lifecycleService.viewDidEnter$.subscribe(() => {
    window.dispatchEvent(new Event('appload'));
  });
}
```

In ALL Views
```ts
constructor(private lifecycleService: LifecycleService) {}

ionViewDidEnter() {
  this.lifecycleService.viewDidEnter$.next();
}
```

**Comment by alexdabast** [comment]
To help for this bug you could call
`    
window.dispatchEvent(new Event('appload'));
`
in your first tab page event ionWillEnterView

**Comment by liamdebeasi** [comment]
Hi everyone,

Here is a dev build with a proposed fix if anyone is interested in testing: `7.4.2-dev.11695831341.191bdf12`

Install Example:

```
npm install @ionic/angular@7.4.2-dev.11695831341.191bdf12
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/28245, and a fix will be available in an upcoming release of Ionic Framework.
